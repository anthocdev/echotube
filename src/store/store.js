import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
