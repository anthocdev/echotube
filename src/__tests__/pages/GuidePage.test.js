import React from "react";
import ReactDOM from "react-dom";
import GuidePage from "../../modules/Guide";
import { Provider } from "react-redux";
import store from "../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Guide Page Tests", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <GuidePage />
          </Provider>
        </BrowserRouter>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <GuidePage />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
  });
});
