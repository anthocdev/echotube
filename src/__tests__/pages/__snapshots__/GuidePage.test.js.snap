// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guide Page Tests Renders Matches Snapshot 1`] = `
<div
  className="guidePage"
>
  <div
    className="guideContent"
  >
    If you have not yet, read our terms of service in the rules section below. 
    <br />
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      href="/rules"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "backgroundColor": "rgba(153, 50, 204, 0.6)",
          "color": "white",
          "margin": "20px",
        }
      }
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Rules
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <br />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "fontSize": "60px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <div
      className="arrow_box"
    >
      Please sign in by clicking the following icon at the top right hand side of the page. 
      <br />
    </div>
  </div>
</div>
`;
